hedera-node/docs/design/app/workflows.md

```diff
@@ -86,4 +86,114 @@ The query workflow consists of the following steps:
    2. **Find response.** If the result of the query is expected (and not only the costs), the actual query is performed.
 
 Depending on what was requested, either the result of the query or the expected costs are returned to the caller.
-If at anytime an error occurs, a response with the error code is returned.
\ No newline at end of file
+If at anytime an error occurs, a response with the error code is returned.
+
+## Handle Workflow
+
+When a platform transaction reaches consensus and needs to be handled, the `HandleWorkflow.handlePlatformTransaction()` 
+is called.
+Few terms that will be used in the following sections:
+1. **Dispatch**: The context needed for executing business logic of a service. 
+All the transactions are handled by creating a Dispatch and executing business logic. This has two implementations 
+- user transactions scope called `UserTxnDispatch`
+- child transactions scope called `ChildDispatch`
+
+#### Dagger Scopes
+
+![Dagger Scopes](images/dagger_scopes.png)
+
+All the objects used while handling the transaction belong to one of the following Dagger scopes. 
+- **Singleton** - Objects that are created once and used for the entire lifecycle of the application.
+Examples include the `NodeInfo` and `WorkingStateAccessor`.
+- **UserTxnScope** - Objects that are created once for platform transaction. 
+Examples include the `Configuration`, `RecordListBuilder` and `TokenContext`.
+Dagger provides all the objects that can be constructed in this scope [here](https://github.com/hashgraph/hedera-services/tree/develop/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/txn/modules) 
+and [UserTxnComponent](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/txn/UserTransactionComponent.java)
+takes all the inputs that are needed to execute the user transaction.
+- **UserDispatchScope** - Objects that are created once for each user transaction that will be dispatched. 
+Examples include the `SingleTransactionRecordBuilder` for user transaction and `FeeContext`.
+Dagger provides all the objects that can be constructed in this scope in [UserDispatchModule](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/dispatch/user/modules/UserDispatchModule.java) and `UserDispatchComponent`.
```

Comment:
Since these are new components introduced from HandleWorkflow refactoring, it would be good to add either more javedoc on each methods of the classes or add some more descriptions here.

Reply:
Which classes do you mean @iwsimon? 
You would like to have javadoc on all Dagger provides and binds methods in `UserDispatchModule` ?

Reply:
I saw only javadoc on the top of all those Dagger classes. If we could add javadoc on each method, it will be helpful.

Reply:
Okay, will do. Most of them should be self explanatory. Thanks!